### PR TITLE
New Google App Menu Feature

### LINKED ISSUE #137: Add Google Account Button
Hello! First of all, thank you for the amazing theme. I noticed that the current theme does not include a Google account button (as shown in the screenshot). Adding this feature would make my workflow much more convenient, as it would allow quicker access to my Google account settings directly from the theme.

Could you consider adding the Google account button to the theme? This small addition would significantly improve usability. Thank you for your consideration!
Screenshot_20241104_101617

### PR DESCRIPTION
**Pull request by itz-rj-here**
Feature Requested by @Nobody9512 
Closes #137 

More modification needed. Requesting @prem-k-r @XengShi to check all the required changes needed. Also Idk how to add a visiblity toggle button on the menu of the extention. I need @prem-k-r's help for it.

### CODE REVIEW

**Comment by itz-rj-here** [comment]
@prem-k-r can you give me a perfect color code for the app menu container?
image

**Comment by prem-k-r** [comment]
> @prem-k-r can you give me a perfect color code for the app menu container? image

```: var(--darkColor-blue);```

**Comment by itz-rj-here** [comment]
> > @prem-k-r can you give me a perfect color code for the app menu container? image
> 
> ```: var(--darkColor-blue);```

🗿🤌
What that means¿

**Comment by itz-rj-here** [comment]
I'm going out now. I will come home after 2 hours.

**Comment by prem-k-r** [comment]
> > @prem-k-r can you give me a perfect color code for the app menu container? image
> 
> fill/background/color`: var(--darkColor-blue);`

Edited

**Comment by itz-rj-here** [comment]
> > @prem-k-r can you give me a perfect color code for the app menu container? image
> 
> fill/background/color`: var(--darkColor-blue);`

Doesn't matches the theme :(
let me show you

**Comment by itz-rj-here** [comment]
@prem-k-r you should add tooltip message on theme selector like this
image

**Comment by itz-rj-here** [comment]
@XengShi Can you tell me something about the scroll bar on the app menu? It's not looking good as I wanted. Do you have any idea?

**Comment by XengShi** [comment]
> @XengShi Can you tell me something about the scroll bar on the app menu? It's not looking good as I wanted. Do you have any idea?

You can see the code of scrollbar that we have for menu.

**Comment by itz-rj-here** [comment]
Uhhhhhhhhh
My brain is not working on these
image

**Comment by XengShi** [comment]
> Uhhhhhhhhh
> My brain is not working on these
> image
> 

These tiles looks great

**Comment by itz-rj-here** [comment]
> > Uhhhhhhhhh
> > My brain is not working on these
> > image
> 
> These tiles looks great

Wait what? 💀

**Comment by itz-rj-here** [comment]
How is it looks now?

image
Red
image
Yellow
image
Green
image
Cyan
image
Blue
image
Pink
image
Purple
image
Orange
image
Wait what. Why this color is full black?
Bruh

**Comment by itz-rj-here** [comment]
@XengShi Bruh change the bg color of the black theme -_-

**Comment by itz-rj-here** [comment]
@prem-k-r can you tell me what is the code to add a visiblity toggle for app menu? Please 🥺

**Comment by itz-rj-here** [comment]
Everything is done but there is one problem. The text color is looking good on the icon color.
image
Somehow we need to setup that, when the black theme is selected the icon color code will change this "color: var(--darkColor-blue);" to this "color: var(--textColorDark-blue);". Can anyone of you give me an idea for this?

**Comment by itz-rj-here** [comment]
Maybe we need to setup this on script.js for this.

**Comment by itz-rj-here** [comment]
The whole world is offline unless me :(

**Comment by XengShi** [comment]
Man have some patience

**Comment by XengShi** [comment]
> @prem-k-r can you tell me what is the code to add a visiblity toggle for app menu? Please 🥺


<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Toggle Div Visibility</title>
<style>
  #myDiv {
    display: none; /* Initially hidden */
    padding: 20px;
    background-color: lightblue;
  }
  
  .visible {
    display: block; /* Make it visible when the class is added */
  }
</style>
</head>
<body>

<button id="toggleButton">Show/Hide Div</button>
<div id="myDiv">This is a toggleable div!</div>

<script>
  const button = document.getElementById('toggleButton');
  const div = document.getElementById('myDiv');

  button.addEventListener('click', () => {
    // Toggle the 'visible' class
    div.classList.toggle('visible');
  });
</script>

</body>
</html>

**Comment by XengShi** [comment]
In css make class :

.visible {
display: block; /* Make it visible when the class is added */
}



Initially set display: none; for menu container.

const menuBTN = document.getElementById('your Menu button's I'd ');
  const div = document.getElementById('Your-menu-container');

  menuBTN.addEventListener('click', () => {
 
    div.classList.toggle('visible');
  });

**Comment by itz-rj-here** [comment]
Couldn't understand :(

**Comment by XengShi** [comment]
> Couldn't understand :(

#noob

**Comment by itz-rj-here** [comment]
> > Couldn't understand :(
> 
> #noob

-_-
I already told ya that I'm a Junior Dev.

**Comment by itz-rj-here** [comment]
But let me still try.

**Comment by XengShi** [comment]
> But let me still try.

Carefully read my previous response you will get that...

**Comment by itz-rj-here** [comment]
This close icon somehow came littlebit of upwards. did i made any mistake?

**Comment by prem-k-r** [comment]
> This close icon somehow came littlebit of upwards. did i made any mistake? image

Probably .icon in css is causing issue. 
Check with `.icon-item .icon {}`

**Comment by itz-rj-here** [comment]
> > This close icon somehow came littlebit of upwards. did i made any mistake? image
> 
> Probably .icon in css is causing issue. Check with `.icon-item .icon {}`

Ohh yeah you're right. Let me try to fix it.

**Comment by itz-rj-here** [comment]
> Everything is done but there is one problem. The text color is looking good on the icon color. image Somehow we need to setup that, when the black theme is selected the icon color code will change this "color: var(--darkColor-blue);" to this "color: var(--textColorDark-blue);". Can anyone of you give me an idea for this?

@XengShi I have to do some studies. Tomorrow is my last exam. After tomorrow I will be free for like 3-4 months. For now, I still don't understand the visibility toggle setup. Just add this code for now and add the toggle setup on the menu. Also, add translation for the text too and do something with the black theme thingy. I'm going for now. Byee. :(

**Comment by prem-k-r** [comment]
Bro, why are you doing all this stuff.. focus on exam first

**Comment by XengShi** [comment]
@itz-rj-here Please please focus on your exams, you write the code later.

BTW I believe '.... a single sheet of paper cannot decide your future' quote from Thomas Edison

**Comment by itz-rj-here** [comment]
> Bro, why are you doing all this stuff.. focus on exam first

Not that much hard thats way. Also I finished my all important exams too.

**Comment by itz-rj-here** [comment]
> @itz-rj-here Please please focus on your exams, you write the code later.
> 
> BTW I believe '.... a single sheet of paper cannot decide your future' quote from Thomas Edison

Bruh -_-
I can finish my tomorrows exam easy. Believe me :(

**Comment by itz-rj-here** [comment]
It's a religion exam anyways. :)

**Comment by itz-rj-here** [comment]
I didn't thoght @XengShi is a bad minded. :)
image

**Comment by itz-rj-here** [comment]
@prem-k-r Actually I'm having a problem on the visibility toggle. (First of all my exam is finished no need to worry now 😀)
As XengShi told me I tried but I'm having problem on adding this cause there is a problem on this.
image
Cause of this styling on html code it's giving me problem and also I still don't understand the thing to add the button on the menu section. 

If you don't mind can you go to my extention repo and open a pr with modifying the menu toggle for the google apps menu please. 😞

**Comment by prem-k-r** [comment]
I'll try, later

**Comment by itz-rj-here** [comment]
Guys I have to go to hospital see you later.

**Comment by XengShi** [comment]
> I didn't thoght @XengShi is a bad minded. :)
> image
> 

What 😯! is this still there ?

**Comment by itz-rj-here** [comment]
> > I didn't thoght @XengShi is a bad minded. :)
> > image
> 
> What 😯! is this still there ?

ofc

**Comment by XengShi** [comment]
@itz-rj-here Should I merge it now?

**Comment by itz-rj-here** [comment]
> @itz-rj-here Should I merge it now?

No no not now

**Comment by itz-rj-here** [comment]
I'm doing some more changes. W8 a little bit time.

**Comment by XengShi** [comment]
Increase the button size (9 dots)

**Comment by itz-rj-here** [comment]
> Increase the button size (9 dots)

What should i keep the width and height?

**Comment by itz-rj-here** [comment]
@XengShi After @prem-k-r helps me with adding the button on the menu section then merge it.

**Comment by XengShi** [comment]
Screenshot from 2024-11-18 11-39-01

For darkmode change 
.menuicon -->    "color: var(--darkColor-blue);"  to  "color: var(--darkerColor-blue);"

**Comment by itz-rj-here** [comment]
> Screenshot from 2024-11-18 11-39-01
> 
> For darkmode change .menuicon --> "color: var(--darkColor-blue);" to "color: var(--darkerColor-blue);"

I done that before but when i change it the other themes color doesn't looks good :(

**Comment by itz-rj-here** [comment]
image
image
image

Here's some examples.

**Comment by XengShi** [comment]
if you change it only for [.menuicon] how other elements can be affected?

**Comment by itz-rj-here** [comment]
> if you change it only for [.menuicon] how other elements can be affected?

-_-
Cause it's all in one kind of. Somehow we need to set the color when selected black the color of this
.menuicon --> "color: var(--darkColor-blue);" to "color: var(--darkerColor-blue);"
changes to this. (Only for black theme)

**Comment by XengShi** [comment]
If you're using SVGs where colors can be easily modified via CSS without adding class names inside the SVG tag, please avoid embedding SVG code directly in the HTML. This helps keep the HTML file clean and organized.

**Comment by prem-k-r** [comment]
@itz-rj-here probably XengShi is asking to change in script.js where dark theme css is defined. Not in style.css

**Comment by itz-rj-here** [comment]
> @itz-rj-here probably XengShi is asking to change in script.js where dark theme css is defined. Not in style.css

Uhhh.
I don't even started learning the js cource. Also I used some help of chatgpt for the js code. I can't do that thing :(

**Comment by itz-rj-here** [comment]
> @itz-rj-here probably XengShi is asking to change in script.js where dark theme css is defined. Not in style.css

Still let me try when selecting black theme the color of "var(--darkColor-blue);" this will change to this "var(--darkerColor-blue);"

**Comment by itz-rj-here** [comment]
@prem-k-r Just do me a favor for me, Add the google apps menu button on the menu section. When toggleing it, it will be visible on the screen.

**Comment by itz-rj-here** [comment]
Do the changes of the button on my repo pr. I will add you to the co-author too. don't worry :)

**Comment by itz-rj-here** [comment]
Uhhh Ok no need When my pr is merged then add your button pr.

**Comment by itz-rj-here** [comment]
@XengShi Merge this pr now. @prem-k-r will add the button on the menu section later on his new pr.

**Comment by itz-rj-here** [comment]
For now let the black theme icon as it is. I will find a solution for it later.

**Comment by prem-k-r** [comment]
> @prem-k-r Just do me a favor for me, Add the google apps menu button on the menu section. When toggleing it, it will be visible on the screen.

bruhh, wait na

**Comment by itz-rj-here** [comment]
> > @prem-k-r Just do me a favor for me, Add the google apps menu button on the menu section. When toggleing it, it will be visible on the screen.
> 
> bruhh, wait na

Okay

**Comment by prem-k-r** [comment]
> @itz-rj-here probably XengShi is asking to change in script.js where dark theme css is defined. Not in style.css

image

``` 
.menuicon{
                color: var(--darkerColor-dark);
            }
```

**Comment by XengShi** [comment]
> If you're using SVGs where colors can be easily modified via CSS without adding class names inside the SVG tag, please avoid embedding SVG code directly in the HTML. This helps keep the HTML file clean and organized.

What about this....?

**Comment by itz-rj-here** [comment]
> > @itz-rj-here probably XengShi is asking to change in script.js where dark theme css is defined. Not in style.css
> 
> image
> 
> ```
> .menuicon{
>                 color: var(--darkerColor-dark);
>             }
> ```

Bruh it was that much ezy? -_-
uhhhhhhhhhh
Gimme your brain 🧠

**Comment by itz-rj-here** [comment]
> > If you're using SVGs where colors can be easily modified via CSS without adding class names inside the SVG tag, please avoid embedding SVG code directly in the HTML. This helps keep the HTML file clean and organized.
> 
> What about this....?

I didn't understand what you wanted to say. :(

**Comment by itz-rj-here** [comment]
@prem-k-r what about the scroll bar is it looking good?

**Comment by prem-k-r** [comment]
> > Increase the button size (9 dots)
> 
> What should i keep the width and height?

width="36" height="36"

also 
.dot-icon-container {
    right: 20px;
  }

**Comment by prem-k-r** [comment]
> @prem-k-r what about the scroll bar is it looking good?

yep it's nice. just that right side border width is uneven with rest of the sides, coz of scrollbar bg is not transarent

**Comment by itz-rj-here** [comment]
> > @prem-k-r what about the scroll bar is it looking good?
> 
> yep it's nice. just that right side border width is uneven with rest of the sides, coz of scrollbar bg is not transarent

So do i need to change something?

**Comment by XengShi** [comment]
Change
```
  <a href="https://google.com/maps" class="icon-item">
            <div class="menuicon">
                <svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" viewBox="0 0 24 24">
                    <path fill="currentColor"
                        d="M16 10V6.85zM4.35 20.7q-.5.2-.925-.112T3 19.75v-14q0-.325.188-.575T3.7 4.8l4.65-1.575Q8.675 3.1 9 3.1t.65.125L15 5.1l4.65-1.8q.5-.2.925.113T21 4.25v7.425q0 .425-.287.713t-.713.287t-.712-.288t-.288-.712V5.7l-3 1.15v2.5q0 .425-.288.713T15 10.35t-.712-.287T14 9.35v-2.5l-4-1.4v12.025q0 .475-.262.85t-.688.55zM5 18.3l3-1.15V5.45l-3 1zm11-.3q.85 0 1.413-.5T18 16q.025-.85-.562-1.425T16 14t-1.425.575T14 16t.575 1.425T16 18m0 2q-1.65 0-2.825-1.175T12 16t1.175-2.825T16 12t2.825 1.175T20 16q0 .575-.137 1.088t-.413.962l1.85 1.85q.275.275.275.7t-.275.7t-.7.275t-.7-.275l-1.85-1.85q-.45.275-.962.413T16 20M8 5.45v11.7z" />
                </svg>
            </div>
            <div class="label short">Maps</div>
        </a>
 
```
to 

```
 <a href="https://google.com/maps" class="icon-item"></a>
            <div class="menuicon">
                <img src="./googleIcons/maps.svg" alt="maps">
            </div>
            <div class="label short">Maps</div>
        </a>
        
       .menuicon img{fill:currentColor} 

```


and see if its working or not

**Comment by prem-k-r** [comment]
https://github.com/user-attachments/assets/02a63dd3-9f0b-4921-a2c8-2ed7308ab1d0

won't it be better if instead of shifting, just show it above below like in google:
image

**Comment by prem-k-r** [comment]
> > > @prem-k-r what about the scroll bar is it looking good?
> > 
> > 
> > yep it's nice. just that right side border width is uneven with rest of the sides, coz of scrollbar bg is not transarent
> 
> So do i need to change something?

.icon-container::-webkit-scrollbar-track {
    background-color: transparent;
  }

File: style.css [hunk 1]
```diff
@@ -355,7 +355,7 @@ body {
 }
 
 .icon-container::-webkit-scrollbar-track {
-  background-color: var(--bg-color-blue); /* Match the border color */
+  background-color: transparent; /* Match the border color */
```
Comment (prem-k-r) [inline style.css:358]: delete comment too lol, doesnt match

Comment (itz-rj-here) [reply style.css:358]: Lol

**Comment by itz-rj-here** [comment]
> Change
> 
> ```
>   <a href="https://google.com/maps" class="icon-item">
>             <div class="menuicon">
>                 <svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" viewBox="0 0 24 24">
>                     <path fill="currentColor"
>                         d="M16 10V6.85zM4.35 20.7q-.5.2-.925-.112T3 19.75v-14q0-.325.188-.575T3.7 4.8l4.65-1.575Q8.675 3.1 9 3.1t.65.125L15 5.1l4.65-1.8q.5-.2.925.113T21 4.25v7.425q0 .425-.287.713t-.713.287t-.712-.288t-.288-.712V5.7l-3 1.15v2.5q0 .425-.288.713T15 10.35t-.712-.287T14 9.35v-2.5l-4-1.4v12.025q0 .475-.262.85t-.688.55zM5 18.3l3-1.15V5.45l-3 1zm11-.3q.85 0 1.413-.5T18 16q.025-.85-.562-1.425T16 14t-1.425.575T14 16t.575 1.425T16 18m0 2q-1.65 0-2.825-1.175T12 16t1.175-2.825T16 12t2.825 1.175T20 16q0 .575-.137 1.088t-.413.962l1.85 1.85q.275.275.275.7t-.275.7t-.7.275t-.7-.275l-1.85-1.85q-.45.275-.962.413T16 20M8 5.45v11.7z" />
>                 </svg>
>             </div>
>             <div class="label short">Maps</div>
>         </a>
>  
> ```
> 
> to
> 
> ```
>  <a href="https://google.com/maps" class="icon-item"></a>
>             <div class="menuicon">
>                 maps
>             </div>
>             <div class="label short">Maps</div>
>         </a>
>         
>        .menuicon img{fill:currentColor} 
> ```
> 
> and see if its working or not

image
It works. Do i replace all now?

**Comment by itz-rj-here** [comment]
> Recording.2024-11-18.124821.mp4 
> won't it be better if instead of shifting, just show it above below like in google: image

W8 let me try

**Comment by prem-k-r** [comment]
for toogle
in html:
```
<!-- 9 Dot Icon -->
    <div class="googleAppsCont" id="googleAppsCont">
``` 
change variable name
.
Add after AI-tools
```
<div class="ttcont">
                            <div class="texts">
                                <div class="bigText" id="ai_tools_button">AI-Tools</div>
                                <div class="infoText" id="enable_ai_tools">Show shortcuts for AI tools</div>
                            </div>
                            <label class="switch">
                                <input id="aiToolsCheckbox" type="checkbox">
                                <span class="toggle"></span>
                            </label>
                        </div>

                        <div class="ttcont">
                            <div class="texts">
                                <div class="bigText" id="googleAppsMenuText">Google Apps</div>
                                <div class="infoText" id="googleAppsMenuInfo">Show shortcut for Google apps</div>
                            </div>
                            <label class="switch">
                                <input id="googleAppsCheckbox" type="checkbox">
                                <span class="toggle"></span>
                            </label>
                        </div>
                        <!-- ---🟡--- -->
```
.
in scrit.js
line 212, change variable name
```
document.querySelector('.googleAppsCont').addEventListener('click', function () {
```
.
image
just follow aiTools, copy paste after them
const googleAppsCont = document.getElementById("googleAppsCont");
const googleAppsCheckbox = document.getElementById("googleAppsCheckbox");
image
```
googleAppsCheckbox.addEventListener("change", function () {
        saveCheckboxState("googleAppsCheckboxState", googleAppsCheckbox);
        if (googleAppsCheckbox.checked) {
            googleAppsCont.style.display = "flex";
            saveDisplayStatus("googleAppsDisplayStatus", "flex");
        } else {
            googleAppsCont.display = "none";
            saveDisplayStatus("googleAppsDisplayStatus", "none");
        }
    });
```
.
image
    loadCheckboxState("googleAppsCheckboxState", googleAppsCheckbox);
    loadDisplayStatus("googleAppsDisplayStatus", googleAppsCont);
.
.
Now css left, just follow what XengShi said, take help from already existed AI-Tools button style property, follow similarly

**Comment by itz-rj-here** [comment]
> for toggle in html:
> 
> ```
> 
>     <div class="googleAppsCont" id="googleAppsCont">
> ```
> 
> change variable name . Add after AI-tools
> 
> ```
> <div class="ttcont">
>                             <div class="texts">
>                                 <div class="bigText" id="ai_tools_button">AI-Tools</div>
>                                 <div class="infoText" id="enable_ai_tools">Show shortcuts for AI tools</div>
>                             </div>
>                             <label class="switch">
>                                 <input id="aiToolsCheckbox" type="checkbox">
>                                 <span class="toggle"></span>
>                             </label>
>                         </div>
> 
>                         <div class="ttcont">
>                             <div class="texts">
>                                 <div class="bigText" id="googleAppsMenuText">Google Apps</div>
>                                 <div class="infoText" id="googleAppsMenuInfo">Show shortcut for Google apps</div>
>                             </div>
>                             <label class="switch">
>                                 <input id="googleAppsCheckbox" type="checkbox">
>                                 <span class="toggle"></span>
>                             </label>
>                         </div>
>                         
> ```
> 
> . in script.js line 212, change variable name
> 
> ```
> document.querySelector('.googleAppsCont').addEventListener('click', function () {
> ```
> 
> line 188 document.getElementById('googleAppsCont').addEventListener('click', function (event) {
> 
> . image just follow aiTools, copy paste after them const googleAppsCont = document.getElementById("googleAppsCont"); const googleAppsCheckbox = document.getElementById("googleAppsCheckbox"); image
> 
> ```
> googleAppsCheckbox.addEventListener("change", function () {
>         saveCheckboxState("googleAppsCheckboxState", googleAppsCheckbox);
>         if (googleAppsCheckbox.checked) {
>             googleAppsCont.style.display = "flex";
>             saveDisplayStatus("googleAppsDisplayStatus", "flex");
>         } else {
>             googleAppsCont.display = "none";
>             saveDisplayStatus("googleAppsDisplayStatus", "none");
>         }
>     });
> ```
> 
> . image loadCheckboxState("googleAppsCheckboxState", googleAppsCheckbox); loadDisplayStatus("googleAppsDisplayStatus", googleAppsCont); . . Now CSS left, just follow what XengShi said, take help from already existed AI-Tools button style property, follow similarly

Got it but it doesn't again hides when disabling the visibility :(
I had to refresh to hide it.
Should i keep as it is or you have other plan?

**Comment by prem-k-r** [comment]
keep it, need to tweak a little in css

**Comment by prem-k-r** [comment]
Nice!
I had thought of doing it along with search engines and GitHub.

**Comment by itz-rj-here** [comment]
I didn't translated the other languages fr.

**Comment by itz-rj-here** [comment]
> > Change
> > ```
> >   <a href="https://google.com/maps" class="icon-item">
> >             <div class="menuicon">
> >                 <svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" viewBox="0 0 24 24">
> >                     <path fill="currentColor"
> >                         d="M16 10V6.85zM4.35 20.7q-.5.2-.925-.112T3 19.75v-14q0-.325.188-.575T3.7 4.8l4.65-1.575Q8.675 3.1 9 3.1t.65.125L15 5.1l4.65-1.8q.5-.2.925.113T21 4.25v7.425q0 .425-.287.713t-.713.287t-.712-.288t-.288-.712V5.7l-3 1.15v2.5q0 .425-.288.713T15 10.35t-.712-.287T14 9.35v-2.5l-4-1.4v12.025q0 .475-.262.85t-.688.55zM5 18.3l3-1.15V5.45l-3 1zm11-.3q.85 0 1.413-.5T18 16q.025-.85-.562-1.425T16 14t-1.425.575T14 16t.575 1.425T16 18m0 2q-1.65 0-2.825-1.175T12 16t1.175-2.825T16 12t2.825 1.175T20 16q0 .575-.137 1.088t-.413.962l1.85 1.85q.275.275.275.7t-.275.7t-.7.275t-.7-.275l-1.85-1.85q-.45.275-.962.413T16 20M8 5.45v11.7z" />
> >                 </svg>
> >             </div>
> >             <div class="label short">Maps</div>
> >         </a>
> >  
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > to
> > ```
> >  <a href="https://google.com/maps" class="icon-item"></a>
> >             <div class="menuicon">
> >                 maps
> >             </div>
> >             <div class="label short">Maps</div>
> >         </a>
> >         
> >        .menuicon img{fill:currentColor} 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and see if its working or not
> 
> image It works. Do i replace all now?

@XengShi didn't replied what should i do now?

**Comment by prem-k-r** [comment]
> > > Change
> > > ```
> > >   <a href="https://google.com/maps" class="icon-item">
> > >             <div class="menuicon">
> > >                 <svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" viewBox="0 0 24 24">
> > >                     <path fill="currentColor"
> > >                         d="M16 10V6.85zM4.35 20.7q-.5.2-.925-.112T3 19.75v-14q0-.325.188-.575T3.7 4.8l4.65-1.575Q8.675 3.1 9 3.1t.65.125L15 5.1l4.65-1.8q.5-.2.925.113T21 4.25v7.425q0 .425-.287.713t-.713.287t-.712-.288t-.288-.712V5.7l-3 1.15v2.5q0 .425-.288.713T15 10.35t-.712-.287T14 9.35v-2.5l-4-1.4v12.025q0 .475-.262.85t-.688.55zM5 18.3l3-1.15V5.45l-3 1zm11-.3q.85 0 1.413-.5T18 16q.025-.85-.562-1.425T16 14t-1.425.575T14 16t.575 1.425T16 18m0 2q-1.65 0-2.825-1.175T12 16t1.175-2.825T16 12t2.825 1.175T20 16q0 .575-.137 1.088t-.413.962l1.85 1.85q.275.275.275.7t-.275.7t-.7.275t-.7-.275l-1.85-1.85q-.45.275-.962.413T16 20M8 5.45v11.7z" />
> > >                 </svg>
> > >             </div>
> > >             <div class="label short">Maps</div>
> > >         </a>
> > >  
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > to
> > > ```
> > >  <a href="https://google.com/maps" class="icon-item"></a>
> > >             <div class="menuicon">
> > >                 maps
> > >             </div>
> > >             <div class="label short">Maps</div>
> > >         </a>
> > >         
> > >        .menuicon img{fill:currentColor} 
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > and see if its working or not
> > 
> > 
> > image
> > It works. Do i replace all now?
> 
> @XengShi didn't replied what should i do now?

Maybe do it. It's better that way.. organized

**Comment by itz-rj-here** [comment]
> > > > Change
> > > > ```
> > > >   <a href="https://google.com/maps" class="icon-item">
> > > >             <div class="menuicon">
> > > >                 <svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" viewBox="0 0 24 24">
> > > >                     <path fill="currentColor"
> > > >                         d="M16 10V6.85zM4.35 20.7q-.5.2-.925-.112T3 19.75v-14q0-.325.188-.575T3.7 4.8l4.65-1.575Q8.675 3.1 9 3.1t.65.125L15 5.1l4.65-1.8q.5-.2.925.113T21 4.25v7.425q0 .425-.287.713t-.713.287t-.712-.288t-.288-.712V5.7l-3 1.15v2.5q0 .425-.288.713T15 10.35t-.712-.287T14 9.35v-2.5l-4-1.4v12.025q0 .475-.262.85t-.688.55zM5 18.3l3-1.15V5.45l-3 1zm11-.3q.85 0 1.413-.5T18 16q.025-.85-.562-1.425T16 14t-1.425.575T14 16t.575 1.425T16 18m0 2q-1.65 0-2.825-1.175T12 16t1.175-2.825T16 12t2.825 1.175T20 16q0 .575-.137 1.088t-.413.962l1.85 1.85q.275.275.275.7t-.275.7t-.7.275t-.7-.275l-1.85-1.85q-.45.275-.962.413T16 20M8 5.45v11.7z" />
> > > >                 </svg>
> > > >             </div>
> > > >             <div class="label short">Maps</div>
> > > >         </a>
> > > >  
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > to
> > > > ```
> > > >  <a href="https://google.com/maps" class="icon-item"></a>
> > > >             <div class="menuicon">
> > > >                 maps
> > > >             </div>
> > > >             <div class="label short">Maps</div>
> > > >         </a>
> > > >         
> > > >        .menuicon img{fill:currentColor} 
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > and see if its working or not
> > > 
> > > 
> > > image
> > > It works. Do i replace all now?
> > 
> > 
> > @XengShi didn't replied what should i do now?
> 
> Maybe do it. It's better that way.. organized

Fine. :)

**Comment by XengShi** [comment]
@itz-rj-here Yes if it's working replace it, doing this will make our html file clean.

And can you please recheck your entire code and make sure IDs and class name you are using are unique and not affecting any existing Id or classes.

**Comment by itz-rj-here** [comment]
> @itz-rj-here Yes if it's working replace it, doing this will make our html file clean.
> 
> And can you please recheck your entire code and make sure IDs and class name you are using are unique and not affecting any existing Id or classes.

I hope so cause when i saw the Close icon issue i completely used other id's

**Comment by itz-rj-here** [comment]
@XengShi Can you tell me why you used the alt on the img code?

**Comment by XengShi** [comment]
> @XengShi Can you tell me why you used the alt on the img code?

Alt is for some reason if the image can't be loaded, html or the browser will show the alt text

**Comment by itz-rj-here** [comment]
> > @XengShi Can you tell me why you used the alt on the img code?
> 
> Alt is for some reason if the image can't be loaded, html or the browser will show the alt text

Oo I see.

**Comment by itz-rj-here** [comment]
@XengShi idk why it's not working now :(

image

**Comment by itz-rj-here** [comment]
Now what should i do?

**Comment by itz-rj-here** [comment]
> Recording.2024-11-18.124821.mp4 
> won't it be better if instead of shifting, just show it above below like in google: image

Idk why but it doesn't place the text under the icons.

**Comment by itz-rj-here** [comment]
image
like this?

**Comment by itz-rj-here** [comment]
Did this
image

But still black. 
image

**Comment by XengShi** [comment]
> image
> like this?

Lol 'currentColor' is not a color it is the class name that you were using previously

**Comment by itz-rj-here** [comment]
> Did this image
> 
> But still black. image

what about this?

**Comment by itz-rj-here** [comment]
Successfully wasted 4 hours on this :)

**Comment by itz-rj-here** [comment]
I'm going for a break.

**Comment by prem-k-r** [comment]
Check with 
.menuicon {
// This and that
}

.menuicon img{
fill: var(--darkColor-blue);
}

**Comment by itz-rj-here** [comment]
You guys got any solution? Or should i revert to old code?

**Comment by prem-k-r** [comment]
I'll check from pc in few minutes

**Comment by prem-k-r** [comment]
I couldn't make it work either. Let's hear from XengShi and then decide whether to revert or have him come up with a solution.

**Comment by prem-k-r** [comment]
> You guys got any solution? Or should i revert to old code?


> I couldn't make it work either. Let's hear from XengShi and then decide whether to revert or have him come up with a solution.

I guess revert it 🫥

**Comment by prem-k-r** [comment]
> Recording.2024-11-18.124821.mp4 
> won't it be better if instead of shifting, just show it above below like in google: image

Btw, have you figured it out?

**Comment by itz-rj-here** [comment]
> > Recording.2024-11-18.124821.mp4
> > won't it be better if instead of shifting, just show it above below like in google: image
> 
> Btw, have you figured it out?

As i said. It doesn't place the text under the downside icons + Also tried with chatgpt and copilot. No solution found.

**Comment by prem-k-r** [comment]
Oh ok

**Comment by prem-k-r** [comment]
@itz-rj-here raised PR in your fork.

Also can you please deploy your fork too so that we can check it from
`https://itz-rj-here.github.io/materialYouNewTab/`
fork repo > settings > pages > Build and deployment

**Review by prem-k-r — APPROVED** [review]
Everything seems fine.

Just that someone can fix this later: The multi-line label pushes the rows below on hover. Instead, it should display the expanded label above the below entity.

**Comment by itz-rj-here** [comment]
Let @XengShi Review it then merge it. :)

**Comment by XengShi** [comment]
> @XengShi Can you check this pr for everything ok?

Everything seems okay, can you initially enable the [google apps] icon ? 
Like when a user adds the extension for the first time icon should be visible/enabled

**Comment by itz-rj-here** [comment]
> > @XengShi Can you check this pr for everything ok?
> 
> Everything seems okay, can you initially enable the [google apps] icon ? Like when a user adds the extension for the first time icon should be visible/enabled

You want it to be visible for the first launch?

**Comment by XengShi** [comment]
হ্যাঁ

**Comment by prem-k-r** [comment]
Shortcuts and AI Tools is also hidden for first time launch

**Comment by itz-rj-here** [comment]
> হ্যাঁ

💀

**Comment by itz-rj-here** [comment]
> Shortcuts and AI Tools is also hidden for first time launch

That was my point.

**Comment by itz-rj-here** [comment]
Bro literally used Bangla for the first time 💀

**Comment by itz-rj-here** [comment]
> Shortcuts and AI Tools is also hidden for first time launch

Can you confirm me this?

**Comment by XengShi** [comment]
> Shortcuts and AI Tools is also hidden for first time launch 

[Ai Tools] don't fit perfectly, especially when shortcuts are enabled. Opening and closing AI tools often feels messy and unorganized. 
@prem-k-r @itz-rj-here we also need to fix that maybe for v2.5*

**Comment by itz-rj-here** [comment]
> > Shortcuts and AI Tools is also hidden for first time launch
> 
> [Ai Tools] don't fit perfectly, especially when shortcuts are enabled. Opening and closing AI tools often feels messy and unorganized. @prem-k-r @itz-rj-here we also need to fix that maybe for v2.5*

Hmm. I seee.
So I do make the Google Apps Visible?

**Comment by XengShi** [comment]
Initially, my code was written so that when AI tools are expanded, shortcuts are hidden but someone changed that.

**Comment by itz-rj-here** [comment]
W8 I don't know to how to visible it -_-

**Comment by itz-rj-here** [comment]
> Initially, my code was written so that when AI tools are expanded, shortcuts are hidden but someone changed that.

I don't understand :(

**Comment by prem-k-r** [comment]
> > Shortcuts and AI Tools is also hidden for first time launch
> 
> [Ai Tools] don't fit perfectly, especially when shortcuts are enabled. Opening and closing AI tools often feels messy and unorganized. @prem-k-r @itz-rj-here we also need to fix that maybe for v2.5*

I'm not entirely sure.
I find unorganised only fold arrow is unfolded and show shortcuts alongwith AI tools shortcuts.. Maybe we should remove the fold arrow, it don't really needed there. 
Is that what you meant?

**Comment by itz-rj-here** [comment]
> > > Shortcuts and AI Tools is also hidden for first time launch
> > 
> > 
> > [Ai Tools] don't fit perfectly, especially when shortcuts are enabled. Opening and closing AI tools often feels messy and unorganized. @prem-k-r @itz-rj-here we also need to fix that maybe for v2.5*
> 
> I'm not entirely sure. I find unorganised only fold arrow is unfolded and show shortcuts alongwith AI tools shortcuts.. Maybe we should remove the fold arrow, it don't really needed there. Is that what you meant?

I Agree with your opinion.

**Pull request merged**